### ISSUE TITLE
Formatter removes required commas from TypeScript type

### ISSUE THREAD

**Issue by shaharmor** [oxc-project/oxc#19621]
# Formatter removes comma separators from single-line object type members, producing invalid TypeScript

## Description

Running `oxfmt` on TypeScript files that contain single-line object types with multiple properties removes the comma (or semicolon) separators between members. This produces syntactically invalid TypeScript that fails type-checking with `TS1005: ';' expected`.

Multi-line object types are not affected, as newlines serve as implicit separators.

## Reproduction

**oxfmt version:** 0.2.0

**.oxfmtrc.json:**
```json
{
  "semi": true,
  "singleQuote": true,
  "tabWidth": 2,
  "trailingComma": "all",
  "printWidth": 100
}
```

**Input (`test.ts`):**
```ts
type AppMessage =
  | { type: 'GET_CAMPAIGNS' }
  | { type: 'SAVE_CAMPAIGN'; payload: { campaign: string } }
  | { type: 'DELETE_CAMPAIGN'; payload: { campaignId: string } }
  | { type: 'GET_GLOBAL_STATE' }
  | { type: 'SET_GLOBAL_STATE'; payload: { state: Partial<string> } }
  | { type: 'GET_POSTING_LOG' }
  | { type: 'ADD_POSTING_LOG_ENTRY'; payload: { entry: string } }
  | { type: 'POST_NOW' };
```

**Run:** `npx oxfmt test.ts`

**Output (broken):**
```ts
type AppMessage = { type: "GET_CAMPAIGNS" } | {
  type: "SAVE_CAMPAIGN"
  payload: { campaign: string }
} | { type: "DELETE_CAMPAIGN" payload: { campaignId: string } } | {
  type: "GET_GLOBAL_STATE"
} | { type: "SET_GLOBAL_STATE" payload: { state: Partial<string> } } | {
  type: "GET_POSTING_LOG"
} | { type: "ADD_POSTING_LOG_ENTRY" payload: { entry: string } } | {
  type: "POST_NOW"
}
```

Note how `{ type: "DELETE_CAMPAIGN" payload: { campaignId: string } }` is missing the comma/semicolon between `type` and `payload` — but only for members that oxfmt keeps on a single line. Multi-line members (like `SAVE_CAMPAIGN`) are fine because newlines act as separators.

**TypeScript errors:**
```
test.ts(4,31): error TS1005: ';' expected.
test.ts(6,32): error TS1005: ';' expected.
test.ts(8,37): error TS1005: ';' expected.
```

## Expected behavior

Single-line object type members should retain comma or semicolon separators between properties:
```ts
{ type: "DELETE_CAMPAIGN", payload: { campaignId: string } }
```

or be broken into multiple lines (where newline acts as separator).
